apps/cowswap-frontend/src/modules/accountProxy/hooks/useCurrentAccountProxy.ts

```diff
@@ -87,18 +90,41 @@ export function useCurrentAccountProxyAddress(): string | undefined {
 }
 
 async function getIsProxySetupValid(
+  chainId: SupportedChainId,
   proxyAddress: string,
   provider: Web3Provider,
   cowShedHooks: CowShedHooks,
 ): Promise<boolean | null> {
+  const providerNetwork = await provider.getNetwork()
+
+  // Skip validation if network mismatch
+  if (providerNetwork.chainId !== chainId) return true
```

Comment:
But here we return `true` if the chain doesn't match. Wouldn't this allow the user to continue and place the order because "the proxy setup is valid"?